### PR TITLE
chore: App router migration - fix metadata for bookings/[status] and add env var for /booking page

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4244, CAL-4245

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Navigate to:

1. `future/bookings/[status]`
2. `future/booking/[uid]`
3. `/bookings/[status]`
4. `/booking/[uid]`

### CODE REVIEW

**Review by Udit-takkar — DISMISSED** [review]
@hbjORbj type checks are failing

**Comment by hbjORbj** [comment]
@Udit-takkar Hey Udit, thanks, the type check is fixed now 🙏

File: apps/web/app/WithEmbedSSR.tsx
Comment (Udit-takkar) [inline apps/web/app/WithEmbedSSR.tsx:21]: Why did you changed this?

Comment (Udit-takkar) [reply apps/web/app/WithEmbedSSR.tsx:21]: I'll wait for @hariombalhara approval on this one

Comment (hbjORbj) [reply apps/web/app/WithEmbedSSR.tsx:21]: `context.res.setHeader` will throw an error in app router.

https://github.com/calcom/cal.com/blob/1edd2558466d98f0768f2553a1324589daee9ccd/apps/web/lib/buildLegacyCtx.tsx#L39-L44

Comment (hariombalhara) [reply apps/web/app/WithEmbedSSR.tsx:21]: This newly created response variable isn't being used anywhere. So, this effectively makes it not work at all.

Have fixed it here
https://github.com/calcom/cal.com/pull/16453#discussion_r1745010098

**Review by hariombalhara — DISMISSED** [review]
Left a comment

**Review by sean-brydon — APPROVED** [review]
LGTM - happy with this

**Pull request merged**
